SQL_Assignment_2.sql

```diff
@@ -0,0 +1,171 @@
+CREATE TABLE Employees (
+    EmployeeID INT PRIMARY KEY,
+    FirstName VARCHAR(50),
+    LastName VARCHAR(50),
+    Department VARCHAR(50),
+    Salary DECIMAL(10, 2),
+    HireDate DATE
+);
+
+INSERT INTO EMPLOYEES VALUES(1,'JOHN', 'DOE','Marketing',62000,'2018-07-24');
+INSERT INTO EMPLOYEES VALUES(2,'ANGEL', 'MARIA','IT',53000,'2020-08-26');
+INSERT INTO EMPLOYEES VALUES(3,'CHRISTIAN', 'BALE','Sales',85000,'2021-05-03');
+INSERT INTO EMPLOYEES VALUES(4,'NIKITHA', 'BOBBY','MARKETING',30000,'2023-06-07');
+INSERT INTO EMPLOYEES VALUES(5,'JR', 'VINI','HR',67800,'2022-01-13');
+INSERT INTO EMPLOYEES VALUES(6,'RONALD', 'SUNNY','INTERN',20000,'2024-03-23');
+
+
+--Retrieve all employees who work in Sales, Marketing, or IT departments.
+
+SELECT *
+FROM EMPLOYEES
+WHERE DEPARTMENT IN('Marketing','IT','SALES');
+
+--Find all employees with salaries ranging from $50,000 to $75,000 (inclusive).
+
+SELECT* 
+FROM EMPLOYEES
+WHERE SALARY BETWEEN 50000 AND 75000;
+
+--List all employees whose last name begins with the letter 'S'.
+
+SELECT * FROM EMPLOYEES
+WHERE LASTNAME LIKE 'S%'
+
+--Display all employees with exactly five letters in their first name.
+
+SELECT * FROM EMPLOYEES
+WHERE LEN(FIRSTNAME)=5;
+
+
+--Find employees whose last name starts with either 'B', 'R', or 'S'.
+
+SELECT * FROM EMPLOYEES
+WHERE LASTNAME LIKE 'B%' OR LASTNAME LIKE 'R%' OR LASTNAME LIKE '%S'
+
+--Retrieve all employees whose first name begins with any letter from 'A' through 'M'.
+
+SELECT * FROM EMPLOYEES
+WHERE LEFT(FIRSTNAME,1) BETWEEN 'A' AND 'M';
+
+--List employees whose last name doesn't start with a vowel (A, E, I, O, U).
+
+SELECT * FROM EMPLOYEES
+WHERE LEFT(LASTNAME,1) NOT IN ('A','E','I','O','U');
+
+--Identify employees earning more than $80,000 annually. 
+
+SELECT *
+FROM EMPLOYEES 
+WHERE SALARY>80000
+
+--Find employees who joined the company before 2020. 
+
+SELECT *
+FROM EMPLOYEES 
+WHERE YEAR(HIREDATE)<2020
+
+-- List all employees not named 'John' (first name).
+
+SELECT *
+FROM EMPLOYEES 
+WHERE FIRSTNAME !='JOHN'
+
+
+--Identify Marketing department employees earning $60,000 or less who were hired after June 30, 2019.
+
+SELECT *
+FROM EMPLOYEES
+WHERE DEPARTMENT='MARKETING' AND (SALARY <=60000 OR HIREDATE > '2019-06-30');
+
+--Find employees whose first name contains the letters 'an' anywhere and ends with 'e'.
+
+SELECT *
+FROM EMPLOYEES
+WHERE  FIRSTNAME LIKE '%AN%' AND FIRSTNAME LIKE '%E';    
```

Comment:
can't we achieve this using only one where condition instead of 2 and condition ?

Reply:
Corrections made.